.github/workflows/workflow.yaml

```diff
@@ -1,177 +1,82 @@
-name: CI / CD
+name: Workflow
 
-on: [push, pull_request]
+on: 
+  push:
+    branches:
+      - 'main'
+      - 'ci-*'
+    tags:
+      - 'v*'
+  pull_request:
 
 jobs:
 
-  # bump-tag:
-  #   if: github.ref == 'refs/heads/main'
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v3
-  #       with:
-  #         fetch-depth: '0'
-  #     - id: bump_version
-  #       name: Bump version and push tag
-  #       uses: anothrNick/github-tag-action@1.46.0
-  #       env:
-  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-  #         WITH_V: true
-  #   outputs:
-  #     new_tag: ${{ steps.bump_version.outputs.new_tag }}
-
-  build-linux:
-    runs-on: ubuntu-latest
+  ci:
+    name: CI
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        platform: [gcc]
-    steps:
-      - uses: actions/checkout@v3
-      - name: Setup Ninja
-        uses: ashutoshvarma/setup-ninja@master
-        with:
-          version: 1.10.2
-      - name: Build
-        env:
-          CC: ${{ steps.vars.outputs.cc }}
-          CXX: ${{ steps.vars.outputs.cxx }}
-          MATRIX_PLATFORM: ${{ matrix.platform }}
-        run: .github/scripts/build-linux.sh
-      - name: Test
-        run: |
-          cd build
-          ctest --verbose
-      - name: Upload artifact
-        if: startsWith(github.ref, 'refs/tags/')
-        uses: actions/upload-artifact@v1
-        with:
-          name: artifacts
-          path: linux-amd64.tar.gz
-
-  build-macos:
-    runs-on: macos-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: Setup Ninja
-        uses: ashutoshvarma/setup-ninja@master
-        with:
-          version: 1.10.2
-      - name: Build
-        run: |
-          .github/scripts/build-macos.sh
-      - name: Test
-        run: |
-          cd build
-          ctest --verbose
-      - name: Upload artifact
-        if: startsWith(github.ref, 'refs/tags/')
-        uses: actions/upload-artifact@v1
-        with:
-          name: artifacts
-          path: darwin-amd64.tar.gz
-
-  build-windows:
-    runs-on: windows-2019
-    strategy:
-      matrix:
-        platform: [x86, amd64]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        platform: ['x86', 'amd64']
+        include:
+          - os: ubuntu-latest
+            script: build-linux.sh
+          - os: macos-latest
+            script: build-macos.sh
+          - os: windows-latest
+            script: build-msvc.ps1
+        exclude:
+          - os: ubuntu-latest
+            platform: x86
+          - os: macos-latest
+            platform: x86
```

Comment:
Doesn't this work?
```yml
      matrix:
        include:
          - os: ubuntu-latest
            script: build-linux.sh
          - os: macos-latest
            script: build-macos.sh
          - os: windows-latest
            script: build-msvc.ps1
            platform: amd64
          - os: windows-latest
            script: build-msvc.ps1
            platform: x86
```

In the SDL project, [we don't use `include` but `platform`](https://github.com/libsdl-org/SDL/blob/19ecb64e0dfea5616798f77d8262715f0d722038/.github/workflows/main.yml#L17-L28).

Reply:
thanks for the platform link, I didn't know you could define whole objects as matrix items. The include idea didn't seem to work, but this did.